File: Estrutura da Dados/Fila/C/fila.c
#include "fila.h"

Fila *newFila(int tam)
{
    Fila *f = (Fila *)malloc(sizeof(Fila));

    if (f == NULL)
        return NULL;

    f->fim = -1;
    f->dados = (int *)malloc(tam * sizeof(int));
    f->capacidade = tam;

    return f;
}

bool isFull(Fila *f)
{
    if (f->fim == f->capacidade - 1)
        return true;
    else
        return false;
}

bool isEmpty(Fila *f)
{
    if (f->fim == -1)
        return true;
    else
        return false;
}

void adicinar(Fila *f,int valor)
{
    if (!isFull(f))
    {
        f->fim++;
        f->dados[f->fim] = valor;
    }
    else
        printf("Fila cheia\n");
}

void removerInicio(Fila *f)
{
    if (!isEmpty(f))
    {

        for (int i = 0; i < f->fim; i++)
            f->dados[i] = f->dados[i + 1];

        f->fim--;
    }
    else
        printf("Fila vazia\n");
}

void freeFila(Fila *f)
{
    free(f->dados);
    free(f);
}

File: Estrutura da Dados/Fila/JAVA/Fila.java
class Fila {

    private int fim;
    private int[] dados;
    private int capacidade;

    public Fila(int tam) {

        this.fim = -1;
        this.dados = new int[tam];
        this.capacidade = tam;

    }

    public boolean isFull() {

        if (fim == capacidade - 1)
            return true;
        else
            return false;

    }

    public boolean isEmpty() {

        if (fim == -1)
            return true;
        else
            return false;

    }

    public void Adicionar(int valor) {
        if (!isFull()) {

            fim++;
            dados[fim] = valor;

        } else
            System.out.println("Fila cheia\n");
    }

    public void removerInicio() {
        if (!isEmpty()) {

            for (int i = 0; i < fim; i++)
                dados[i] = dados[i + 1];

            fim--;

        } else
            System.out.println("Fila vazia\n");
    }

}